src/fs/filesystem.ts

```diff
@@ -13,54 +13,75 @@ export class FileSystem {
   privateTree: PrivateTree
   syncHooks: Array<SyncHook>
 
   private key: string
 
-  constructor(root: PublicTree, publicTree: PublicTree, privateTree: PrivateTree, key: string) {
+  constructor(
+    root: PublicTree,
+    publicTree: PublicTree,
+    prettyTree: PublicTree,
+    privateTree: PrivateTree,
+    key: string
+  ) {
     this.root = root
     this.publicTree = publicTree
+    this.prettyTree = prettyTree
     this.privateTree = privateTree
     this.key = key
     this.syncHooks = []
   }
 
   static async empty(opts: FileSystemOptions = {}): Promise<FileSystem> {
     const { keyName = 'filesystem-root', version = semver.latest } = opts
+
     const root = await PublicTree.empty(version)
     const publicTreeInstance = await PublicTree.empty(version)
+    const prettyTreeInstance = await PublicTree.empty(semver.v0)
+
     const privateTreeInstance = await PrivateTree.empty(version)
     const key = await keystore.getKeyByName(keyName)
-    return new FileSystem(root, publicTreeInstance, privateTreeInstance, key)
+
+    return new FileSystem(root, publicTreeInstance, prettyTreeInstance, privateTreeInstance, key)
   }
 
   static async fromCID(cid: CID, opts: FileSystemOptions = {}): Promise<FileSystem | null> {
     const { keyName = 'filesystem-root' } = opts
+
     const root = await PublicTree.fromCID(cid)
     const publicTree = (await root.getDirectChild('public')) as PublicTree
+    const prettyTree = (await root.getDirectChild('pretty')) as PublicTree ||
+                        await PublicTree.empty(semver.v0)
+
     const privLink = root.findLink('private')
     const key = await keystore.getKeyByName(keyName)
     const privateTree = privLink ? await PrivateTree.fromCIDWithKey(privLink.cid, key) : null
-    if (publicTree === null || privateTree === null) {
-      return null
-    }
-    return new FileSystem(root, publicTree, privateTree, key)
+
+    if (publicTree === null || privateTree === null) return null
+
+    return new FileSystem(root, publicTree, prettyTree, privateTree, key)
   }
 
   static async forUser(username: string, opts: FileSystemOptions = {}): Promise<FileSystem | null> {
     const cid = await user.fileRoot(username)
     return FileSystem.fromCID(cid, opts)
   }
 
-  // Upgrade public IPFS folder to FileSystem
+  /**
+   * Upgrade public IPFS folder to FileSystem
+   */
   static async upgradePublicCID(cid: CID, opts: FileSystemOptions = {}): Promise<FileSystem> {
     const { keyName = 'filesystem-root', version = semver.latest } = opts
+
     const root = await PublicTree.empty(version)
     const pubTreeInstance = await PublicTree.fromCID(cid)
+    const pretTreeInstance = await PublicTree.fromCID(cid, semver.v0)
     const privTreeInstance = await PrivateTree.empty(version)
+
     const key = await keystore.getKeyByName(keyName)
-    return new FileSystem(root, pubTreeInstance, privTreeInstance, key)
+    return new FileSystem(root, pubTreeInstance, pretTreeInstance, privTreeInstance, key)
```

Comment:
Let's just type out the words. Yes, it'll be longer, but more readable 4 months from now when we have to dig into this fresh again.

Also FWIW, I don't think we need `Instance` on every one of these.

```js
return new FileSystem(root, publicTree, prettyTree, privateTree, key)
```

Also can't hurt to make the constructor take a map for named arguments.

Reply:
> Also FWIW, I don't think we need Instance on every one of these.

Why not? 🤔 

> Also can't hurt to make the constructor take a map for named arguments.

Agreed 👍